avocado/utils/network/interfaces.py

```diff
@@ -128,6 +128,74 @@ def add_ipaddr(self, ipaddr, netmask):
         except Exception as ex:
             raise NWException("Failed to add address {}".format(ex))
 
+    @property
+    def vlans(self):
+        """Return a list of vlans"""
+        vlans = {}
+        with open('/proc/net/vlan/config', encoding="utf-8") as vlan_config_file:
+            for line in vlan_config_file:
+                # entry is formatted as "vlan_name | vlan_id | parent_device"
+                line = line.strip().replace(" ", "")
+                if line.endswith(self.name):
+                    line = line.split('|')
+                    vlans[line[1]] = line[0]
```

Comment:
IIUC, `line[1]` will be a `str`... but on `remove_vlan_by_tag()`, you do:

```python
if int(vlan_num) in self.vlans.keys()
```

I'm missing how the keys on `self.vlan` will be integers... Also, you can remove the `keys()` when checking for presence, that is:

```python
if int(vlan_num) in self.vlans:
```

Reply:
converted to str instead of int, thanks for pointing this out